### PR TITLE
Adding more validation FA Volume detach test

### PR DESCRIPTION
**Pull request by lsrinivas-pure**
**What this PR does / why we need it**:
Added validation of reattach volume to parent node once failover.
**Which issue(s) this PR fixes** (optional)
Closes #PTX-20620

**Special notes for your reviewer**:
https://aetos.pwx.purestorage.com/resultSet/testSetID/442773

### CODE REVIEW

File: tests/basic/pure_test.go [hunk 1]
```diff
@@ -572,49 +590,78 @@ var _ = Describe("{FADARemoteDetach}", func() {
...
+				log.FailOnError(err, "error enabling scheduling on node [%s]", podNode.Name)
+
+			})
+		}
+
+		validatePodRemoteDetach()
+
+		stepLog = fmt.Sprintf("Schedule pod back on node [%s]", podNode.Name)
+		Step(stepLog, func() {
+			disableSchedulingOnPodNode()
```
Comment (kphalgun-px) [inline tests/basic/pure_test.go:661]: log.InfoD(stepLog)?

File: tests/basic/pure_test.go [hunk 2]
```diff
@@ -572,49 +590,78 @@ var _ = Describe("{FADARemoteDetach}", func() {
...
+
+			return nil, false, nil
+		}
+
+		//Validating  new pod ready  and old pod termination
+		validatePodRemoteDetach := func() {
+
+			stepLog = fmt.Sprintf("Stop Portworx on node [%s] and validate new pod", podNode.Name)
+			Step(stepLog, func() {
+				StopVolDriverAndWait([]node.Node{podNode})
```
Comment (kphalgun-px) [inline tests/basic/pure_test.go:640]: log.InfoD(stepLog)?

Comment (lsrinivas-pure) [reply tests/basic/pure_test.go:635]: fixed

File: tests/basic/pure_test.go (hunk 1 above)
Comment (lsrinivas-pure) [reply tests/basic/pure_test.go:656]: fixed

File: tests/basic/pure_test.go
Comment (kphalgun-px) [inline tests/basic/pure_test.go:1]: line 575:

log.FailOnError(fmt.Errorf("new pod with name [%s] is not *available", appPodName), "error getting new app pod")

File: tests/basic/pure_test.go [hunk 3]
```diff
@@ -572,49 +587,76 @@ var _ = Describe("{FADARemoteDetach}", func() {
...
-			_, err = task.DoRetryWithTimeout(waitForPodTerminated, 5*time.Minute, 10*time.Second)
+			for _, p := range appPods.Items {
+				if p.Name == appPod.Name {
+					return nil, true, fmt.Errorf("pod [%s] still not terminated. Current state [%v]", appPod.Name, p.Status.ContainerStatuses[0].State)
+				}
+			}
+			return nil, false, nil
+		}
+
+		//Validating  new pod ready  and old pod termination
```
Comment (kphalgun-px) [inline tests/basic/pure_test.go:630]: and *org/original? pod termination

File: tests/basic/pure_test.go [hunk 4]
```diff
@@ -536,18 +534,35 @@ var _ = Describe("{FADARemoteDetach}", func() {
...
 			log.InfoD(stepLog)
+			nsList, err := core.Instance().ListNamespaces(map[string]string{"creator": "torpedo"})
+			log.FailOnError(err, "error getting all namespaces")
+			for _, ns := range nsList.Items {
+				if strings.Contains(ns.Name, "fadavoldetach") {
+					appNamespace = ns.Name
+					break
+				}
+			}
+			log.Infof("App deployed in namespace %s", appNamespace)
```
[the remark this answers is not in the document]
Comment (lsrinivas-pure) [reply tests/basic/pure_test.go:550]: we can do that, but doesn't make much diff as we have loop through contextx in that case

File: tests/basic/pure_test.go
Comment (lsrinivas-pure) [reply tests/basic/pure_test.go:1]: done

File: tests/basic/pure_test.go (hunk 3 above)
Comment (lsrinivas-pure) [reply tests/basic/pure_test.go:630]: done

**Review by aravindranath14 — APPROVED** [review]
LGTM

**Pull request merged**